__tests__/components/ResourceCreation/TestResourceCreation.test.tsx

```diff
@@ -0,0 +1,204 @@
+
+describe('TestResourceCreation', () => {
+  it('should not render modal by default', () => {
+    const MockResources = getMockRecoilState(patientTestCaseState, {
+      'example-test-case': {
+        patient: {
+          resourceType: 'Patient',
+          name: [{ given: ['Test123'], family: 'Patient456' }]
+        },
+        resources: []
+      }
+    });
+    const MockSelectedDataRequirement = getMockRecoilState(selectedDataRequirementState, {
+      name: '',
+      content: null
+    });
+    const MockSelectedPatient = getMockRecoilState(selectedPatientState, 'example-test-case');
+
+    render(
+      mantineRecoilWrap(
+        <>
+          <MockResources />
+          <MockSelectedDataRequirement />
+          <MockSelectedPatient />
+          <TestResourceCreation />
+        </>
+      )
+    );
+
+    const modal = screen.queryByTestId('code-editor-modal');
+    expect(modal).not.toBeInTheDocument();
+  });
+
+  it('should render modal when data element is selected from test resource display', () => {
+    const TEST_DATA_REQ: fhir4.DataRequirement = {
+      type: 'Observation',
+      codeFilter: [],
+      dateFilter: []
+    };
+
+    const MockResources = getMockRecoilState(patientTestCaseState, {
+      'example-test-case': {
+        patient: {
+          resourceType: 'Patient',
+          name: [{ given: ['Test123'], family: 'Patient456' }]
+        },
+        resources: []
+      }
+    });
+    const MockSelectedDataRequirement = getMockRecoilState(selectedDataRequirementState, {
+      name: '',
+      content: TEST_DATA_REQ
+    });
+    const MockSelectedPatient = getMockRecoilState(selectedPatientState, 'example-test-case');
+
+    render(
+      mantineRecoilWrap(
+        <>
+          <MockResources />
+          <MockSelectedDataRequirement />
+          <MockSelectedPatient />
+          <TestResourceCreation />
+        </>
+      )
+    );
+
+    const modal = screen.getByTestId('code-editor-modal');
+    expect(modal).toBeInTheDocument();
+  });
+
+  it('should render modal when Edit Resource button is clicked', () => {
+    const MockResources = getMockRecoilState(patientTestCaseState, {
+      'example-test-case': {
+        patient: {
+          resourceType: 'Patient',
+          name: [{ given: ['Test123'], family: 'Patient456' }]
+        },
+        resources: [
+          {
+            resourceType: 'Procedure',
+            id: 'test-id'
+          }
+        ]
+      }
+    });
+    const MockSelectedPatient = getMockRecoilState(selectedPatientState, 'example-test-case');
+
+    render(
+      mantineRecoilWrap(
+        <>
+          <MockResources />
+          <MockSelectedPatient />
+          <TestResourceCreation />
+        </>
+      )
+    );
+
+    const editResourceButton = screen.getByText(/edit fhir resource/i);
+    fireEvent.click(editResourceButton);
+
+    const modal = screen.getByTestId('code-editor-modal');
```

Comment:
```suggestion
    const modal = screen.getByRole('dialog', { name: 'Edit FHIR Resource' });
```
Similar to above, but can be done with modals too. This also allows you to not need to add testids to the resources. The testid method for unit testing should only be used as a last resort.